### DISCUSSION TITLE
Allow for always-on filter labels for colorblind people

### DISCUSSION THREAD

**Discussion by christopherarter** [spatie/ray#110] [Ideas]
I have Deuteranomaly colorblindness and the labels are a bit tough to work with. I see there is a label in the tool tip on hover, but some ability to toggle these to stay always-on below the label would be nice, or some way to rely on text instead of colors. 
<img width="172" alt="Screen Shot 2021-01-11 at 10 19 59 AM" src="https://user-images.githubusercontent.com/12786153/104201089-5e5b4780-53f7-11eb-8759-4ccc5f3f9db9.png">

Other than that, loving Ray so far!

**Comment by AdrianMrn** [comment]
@willemvb Do you have any ideas how we could make this better accessibility-wise?

Since we're thinking about adding more advanced filters, I'm think we could incorporate a "colorblind" setting in the preferences. Maybe for now, a simple multiselect dropdown with the names of the colors could be sufficient?

**Reply by christopherarter** [reply]
Perhaps something like this, with the background fill indicating toggle on / off. But the key being that the text is there as well.

![Screen Shot 2021-01-11 at 11 03 11 AM](https://user-images.githubusercontent.com/12786153/104205635-9e70f900-53fc-11eb-815c-a52fffe44fcd.png)

**Reply by willemvb** [reply]
Thanks for the input @christopherarter! I totally get that the minimal toggles are hard to use sometimes.

Adding a setting and mentioning the names inside the labels looks indeed like a good take on this. We'll have to test further how this would work on smaller windows with a long Ray screen name on the left.

Once we tackle the toggles, we could also look at the actual log items and add the color in text there as well, eg. as an extra label.

@AdrianMrn Ping me when you have time to add the preference, I'll look into the label styling.

**Reply by christopherarter** [reply]
Great, thanks for considering this. I'm enjoying Ray. Keep up the great work!
